config/experimental/configsource/component.go

```diff
@@ -65,7 +67,7 @@ type Session interface {
 	//
 	// The selector is a string that is required on all invocations, the params are optional. Each
 	// implementation handles the generic params according to their requirements.
-	Retrieve(ctx context.Context, selector string, params interface{}) (Retrieved, error)
+	Retrieve(ctx context.Context, selector string, paramsParser *configparser.Parser) (Retrieved, error)
```

Comment:
Very confused about this change, can you elaborate a bit?

Reply:
Two reasons for that change:

1. In practice, config sources, that used interesting parameters had to parse them somehow. Using the `configparser.Parser` keeps the consistency and avoids bringing extra dependencies.

2. For the "merge" of sources as planned to be used by AWS this makes the config sources composable by allowing the params (or parts of them) to come from other config sources that are also handled via the manager. 

This is still a bit abstract - I will add an example here to clarify it.

Reply:
Here is an example trying to convey how this is useful. It makes creating config sources that operate on data from other config sources. In the example below the config source called `meta_config_source` can be used to merge data from multiple other config sources.

```yaml
# Inserting the configurations via different files and merging then.
receivers: |
  $meta_config_source: merge
  sources:
    - $include:./testdata/config_1_for_container_insights.yaml
    - $include:./testdata/config_2_for_app_metrics.yaml
```

Reply:
I think what you want is different for AWS use-case, probably instead of "Retrived.Value()" you need a "Retrived.MergeValue(Parser, key)" to merge the value in the parser at the specified "key"?

Reply:
@tigrannajaryan 
> In the example with $meta_config_source what does : merge part mean? Which one is the type of the config source here? merge or meta_config_source?

The config source in this case is the `meta_config_source` and `merge` is the selector.

`paramsParser` is being renamed to follow other refactors on the code base, now it should be named `paramsConfigMap`. It represents the parameters defined in the configuration that will be passed to the ` Retrieve` operation. On the example above two config maps from the `include`  config source.

@bogdandrutu
> I think what you want is different for AWS use-case, probably instead of "Retrived.Value()" you need a "Retrived.MergeValue(Parser, key)" to merge the value in the parser at the specified "key"?

Yes, having it at the `Retrieved` makes perfect sense. The "meta config source" (or something similar) then can operate at a very high level. That said it still helps to have the params of the Retrieve as config map that makes easier to have have config sources easily manipulating the contents returned by other config sources.